<launch>

    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find espeleo_planner)/config/general.config"/>

    <include file="$(find espeleo_control)/launch/vector_field_action.launch" />
    <!-- <include file="$(find espeleo_lego_loam)/launch/espeleo_lego_loam_sim.launch" /> -->

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="0.2" />
        <param name="publish_free_space" value="true" />
        <param name="frame_id" type="string" value="os1_init" />
        <param name="sensor_model/max_range" value="-1.0" />
        <param name="latch" value="true"/>
        <remap from="cloud_in" to="/full_cloud_projected2" />
    </node>

    <node pkg="espeleo_planner" type="republish_cloud_to_frame"
            name="republish_cloud_to_frame">
    </node>

    <!-- Convert nav_msgs/Path message to Espeleo Path for autonomous navigation -->
    <node pkg="espeleo_planner" type="convert_nav_msgs_path_to_espeleo_path.py"
            name="convert_nav_msgs_path_to_espeleo_path" output="screen">
    </node>

<!--     <include file="$(find espeleo_planner)/launch/mesh_planner_exploration.launch" /> -->

<!--     <node pkg="espeleo_planner" type="exploration_metrics.py" -->
<!--             name="exploration_metrics" output="screen"> -->
<!--     </node> -->

<!--     <node name="rviz" pkg="rviz" type="rviz" -->
<!--         args="-d $(find espeleo_planner)/rviz/espeleo_coppelia_white_PCLOUD.rviz" required="true" /> -->

<!--     <group ns="gt_map"> -->
<!--         <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_gt"> -->
<!--             <param name="resolution" value="0.2" /> -->
<!--             <param name="publish_free_space" value="true" /> -->
<!--             <param name="frame_id" type="string" value="world" /> -->
<!--             <param name="sensor_model/max_range" value="-1.0" /> -->
<!--             <param name="latch" value="true"/> -->
<!--             <remap from="cloud_in" to="/laser_cloud_surround2" /> -->
<!--         </node> -->
<!--     </group> -->
</launch>